packages/graphql-testing/src/createGraphQLFactory.ts

```diff
@@ -0,0 +1,131 @@
+import {GraphQLRequest, ApolloLink, Operation, Observable} from 'apollo-link';
+import {
+  ApolloReducerConfig,
+  InMemoryCache,
+  IntrospectionFragmentMatcher,
+} from 'apollo-cache-inmemory';
+import ApolloClient from 'apollo-client';
+
+import MockApolloLink from './MockApolloLink';
+import Requests from './Requests';
+import {GraphQLMock} from './types';
+
+export interface Options {
+  unionOrIntersectionTypes?: any[];
+  cacheOptions?: ApolloReducerConfig;
+}
+
+export interface GraphQLClientOptions {
+  ssrMode?: boolean;
+}
+
+function defaultGraphQLMock({operationName}: GraphQLRequest) {
+  return new Error(
+    `Can’t perform GraphQL operation '${operationName ||
+      ''}' because no mocks were set.`,
+  );
+}
+
+export type MockGraphQLClient = ApolloClient<unknown> & {
```

Comment:
I still don't think this feels right. Can we please instead subclass `ApolloClient`, and have that subclass do all the work of having an instance property for the operations/ automatically add the `mockLink`?

Reply:
I did half of this. It does not quite allow me to move `memoryLink` to the `MockGraphQLClient` class since super had to be call right away.

Unless I miss-understood what needed to be done?